packages/@aws-cdk/aws-iot-actions-alpha/lib/step-functions-start-state-machine-action.ts

```diff
@@ -0,0 +1,65 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as iot from '@aws-cdk/aws-iot-alpha';
+import * as stepfunctions from 'aws-cdk-lib/aws-stepfunctions';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+import { ArnFormat, Stack } from 'aws-cdk-lib';
+
+/**
+ * Configuration properties of an action for the Step Functions State Machine.
+ */
+export interface StepFunctionsStartStateMachineActionProps extends CommonActionProps {
+  /**
+   * Name of the state machine execution prefix.
+   * The name given to the state machine execution consists of this prefix followed by a UUID. Step Functions creates a unique name for each state machine execution if one is not provided.
+   *
+   * @see https://docs.aws.amazon.com/iot/latest/developerguide/stepfunctions-rule-action.html#stepfunctions-rule-action-parameters
+   *
+   * @default: None
+   */
+  readonly executionNamePrefix?: string;
+}
+
+/**
+ * The action to put the record from an MQTT message to the Step Functions State Machine.
+ */
+export class StepFunctionsStartStateMachineAction implements iot.IAction {
+  private readonly executionNamePrefix?: string;
+  private readonly role?: iam.IRole;
+
+  /**
+   * @param stateMachine The Step Functions Start Machine which shoud be executed.
+   * @param props Optional properties to not use default
+   */
+  constructor(private readonly stateMachine: stepfunctions.IStateMachine, props?: StepFunctionsStartStateMachineActionProps) {
+    this.executionNamePrefix = props?.executionNamePrefix;
+    this.role = props?.role;
+  }
+
+  /**
+   * @internal
+   */
+  public _bind(rule: iot.ITopicRule): iot.ActionConfig {
+    const role = this.role ?? singletonActionRole(rule);
+    const stateMachineName = Stack.of(this.stateMachine).splitArn(this.stateMachine.stateMachineArn, ArnFormat.COLON_RESOURCE_NAME).resourceName;
```

Comment:
Can the user also not pass in this value? https://docs.aws.amazon.com/iot/latest/developerguide/stepfunctions-rule-action.html#stepfunctions-rule-action-parameters

Reply:
StateMachineName is required, see [CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-iot-topicrule-stepfunctionsaction.html#aws-properties-iot-topicrule-stepfunctionsaction-properties). Logically, this attribute is also required - we want to add a new State Machine action. So a state machine has to be specified.